### PR TITLE
Request Validator changes to make contents list as optional and Cassandra migration changes

### PR DESCRIPTION
**Pull request by sowmya-dixit**
@maheshkumargangula Please review

### CODE REVIEW

File: sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.79_cassandra.cql [hunk 1]
```diff
@@ -1,5 +1,5 @@
-DROP TABLE sunbird_courses.assessment_aggregator;
-DROP TYPE sunbird_courses.question;
+DROP TABLE IF EXISTS sunbird_courses.assessment_aggregator;
```
Comment (manzarul) [inline sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.79_cassandra.cql]: Is this file already run on dev then it won't again run , is this ok for you.

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java [hunk 2]
```diff
@@ -40,62 +40,59 @@ private RequestValidator() {}
...
-    if (((List<Map<String, Object>>) (contentRequestDto.getRequest().get(JsonKey.CONTENTS)))
-        .isEmpty()) {
-      throw new ProjectCommonException(
-          ResponseCode.contentIdRequired.getErrorCode(),
-          ResponseCode.contentIdRequiredError.getErrorMessage(),
-          ERROR_CODE);
-    } else {
+
+    List<Map<String, Object>> contents =  (List<Map<String, Object>>) contentRequestDto.getRequest().get(JsonKey.CONTENTS);
+    if (!CollectionUtils.isEmpty(contents)) {
```
Comment (manzarul) [inline sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: CollectionUtils .isNotEmpty use this

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java [hunk 3]
```diff
@@ -40,62 +40,59 @@ private RequestValidator() {}
...
       List<Map<String, Object>> list =
-          (List<Map<String, Object>>) (contentRequestDto.getRequest().get(JsonKey.CONTENTS));
+              (List<Map<String, Object>>) (contentRequestDto.getRequest().get(JsonKey.CONTENTS));
```
Comment (manzarul) [inline sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: if list is null then what will happen

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java [hunk 4]
```diff
@@ -40,62 +40,59 @@ private RequestValidator() {}
...
-              ProjectUtil.isDateValidFormat(
-                  "yyyy-MM-dd HH:mm:ss:SSSZ", (String) map.get(JsonKey.LAST_COMPLETED_TIME));
+                  ProjectUtil.isDateValidFormat(
+                          "yyyy-MM-dd HH:mm:ss:SSSZ", (String) map.get(JsonKey.LAST_COMPLETED_TIME));
           if (!bool) {
             throw new ProjectCommonException(
-                ResponseCode.dateFormatError.getErrorCode(),
-                ResponseCode.dateFormatError.getErrorMessage(),
-                ERROR_CODE);
+                    ResponseCode.dateFormatError.getErrorCode(),
```
Comment (manzarul) [inline sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: dateFormatError can be in one place , as of now same code is above as well.

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java [hunk 5]
```diff
@@ -40,62 +40,59 @@ private RequestValidator() {}
...
+                    ResponseCode.contentStatusRequired.getErrorMessage(),
+                    ERROR_CODE);
           }
 
         } else {
           throw new ProjectCommonException(
-              ResponseCode.contentIdRequired.getErrorCode(),
-              ResponseCode.contentIdRequiredError.getErrorMessage(),
-              ERROR_CODE);
+                  ResponseCode.contentIdRequired.getErrorCode(),
```
Comment (manzarul) [inline sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: contentIdRequired exception can be common for above code

File: sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.79_cassandra.cql (hunk 1 above)
Comment (RevathiKotla) [reply sunbird-cassandra-migration/cassandra-migration/src/main/resources/db/migration/cassandra/V1.79_cassandra.cql]: yes we already tested this in dev

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java (hunk 2 above)
Comment (sowmya-dixit) [reply sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: Done

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java (hunk 3 above)
Comment (sowmya-dixit) [reply sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: CollectionUtils .isNotEmpty will handle null

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java (hunk 4 above)
Comment (sowmya-dixit) [reply sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: This was existing code. We didn't change anything.

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java (hunk 5 above)
Comment (sowmya-dixit) [reply sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: This was existing code. We didn't change anything.

File: sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java (hunk 4 above)
Comment (manzarul) [reply sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/RequestValidator.java]: but it can be updated(:

**Pull request merged**